src/main/java/uk/gov/pay/directdebit/webhook/gocardless/services/WebhookGoCardlessService.java

```diff
@@ -63,22 +72,41 @@ public void handleEvents(List<GoCardlessEvent> events) {
 
     private void updateStateForMandateEvents(List<GoCardlessEvent> eventsThatAffectMandates) {
         eventsThatAffectMandates.stream()
-                .map(WebhookGoCardlessService::toGoCardlessMandateIdAndOrganisationId)
+                .map(this::toMandateBusinessKey)
                 .flatMap(Optional::stream)
                 .distinct()
+                .map(this::getMandate)
+                .flatMap(Optional::stream)
                 .forEach(goCardlessMandateStateUpdater::updateState);
     }
 
-    private static Optional<GoCardlessMandateIdAndOrganisationId> toGoCardlessMandateIdAndOrganisationId(GoCardlessEvent goCardlessEvent) {
+    private Optional<Pair<GoCardlessMandateId, GoCardlessOrganisationId>> toMandateBusinessKey(GoCardlessEvent goCardlessEvent) {
         var goCardlessMandateIdAndOrganisationId = goCardlessEvent.getLinksMandate()
-                .map(mandateId -> new GoCardlessMandateIdAndOrganisationId(mandateId, goCardlessEvent.getLinksOrganisation()))
+                .map(mandateId -> Pair.of(mandateId, goCardlessEvent.getLinksOrganisation()))
```

Comment:
I think I’d prefer to have a specific type, even if it’s used just once, rather than using a `Pair`